In [69]:
# Load the Pandas libraries with alias 'pd' 
import pandas as pd 
# Read data from file 'filename.csv' 
# (in the same directory that your python process is based)
# Control delimiters, rows, column names with read_csv (see later) 
data = pd.read_csv("avocado.csv") 
# Preview the first 5 lines of the loaded data 
data.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

In [70]:
#cantidad de datos que se tienen:
data.count()

Unnamed: 0      18249
Date            18249
AveragePrice    18249
Total Volume    18249
4046            18249
4225            18249
4770            18249
Total Bags      18249
Small Bags      18249
Large Bags      18249
XLarge Bags     18249
type            18249
year            18249
region          18249
dtype: int64

In [71]:
#identifica el tipo de variabesl 
df.dtypes

Unnamed: 0        int64
Date             object
AveragePrice    float64
Total Volume    float64
4046            float64
4225            float64
4770            float64
Total Bags      float64
Small Bags      float64
Large Bags      float64
XLarge Bags     float64
type             object
year              int64
region           object
dtype: object

In [72]:

df = pd.DataFrame(data)
df.max()

Unnamed: 0                    52
Date                  2018-03-25
AveragePrice                3.25
Total Volume         6.25056e+07
4046                 2.27436e+07
4225                 2.04706e+07
4770                 2.54644e+06
Total Bags           1.93731e+07
Small Bags           1.33846e+07
Large Bags            5.7191e+06
XLarge Bags               551694
type                     organic
year                        2018
region          WestTexNewMexico
dtype: object

In [73]:
df.min()

Unnamed: 0                 0
Date              2015-01-04
AveragePrice            0.44
Total Volume           84.56
4046                       0
4225                       0
4770                       0
Total Bags                 0
Small Bags                 0
Large Bags                 0
XLarge Bags                0
type            conventional
year                    2015
region                Albany
dtype: object

In [74]:
#Obtener la media para cada variable
df.mean()

Unnamed: 0          24.232232
AveragePrice         1.405978
Total Volume    850644.013009
4046            293008.424531
4225            295154.568356
4770             22839.735993
Total Bags      239639.202060
Small Bags      182194.686696
Large Bags       54338.088145
XLarge Bags       3106.426507
year              2016.147899
dtype: float64

In [75]:
#Obtener la mediana
df.median()

Unnamed: 0          24.00
AveragePrice         1.37
Total Volume    107376.76
4046              8645.30
4225             29061.02
4770               184.99
Total Bags       39743.83
Small Bags       26362.82
Large Bags        2647.71
XLarge Bags          0.00
year              2016.00
dtype: float64

In [76]:
#obtener la desviación estándar
df.std()

Unnamed: 0      1.548104e+01
AveragePrice    4.026766e-01
Total Volume    3.453545e+06
4046            1.264989e+06
4225            1.204120e+06
4770            1.074641e+05
Total Bags      9.862424e+05
Small Bags      7.461785e+05
Large Bags      2.439660e+05
XLarge Bags     1.769289e+04
year            9.399385e-01
dtype: float64

¿Qué conclusiones puedes entregar de los datos?
De acuerdo a una investigación se tiene que para las columnas  4046, 4225, 4770 se refieren a los siguinetes tamaños de aguacate Hass:
Small/Medium (~3-5oz avocado)= 4046 
Large(~8-10oz avocado)=4225
Extra Large(~10-15oz avocado)=4770 
Bags=Size varies

Entonces dado lo anterior como un marco de referencia, se tiene un precio promedio de venta de 1.37 para estos tipos de aguacate hass, pero que puede alcanzar hasta un precio promedio máximo de 3.25.

Por otro lado, para las bolsas se tiene que las que más se producen son las XLarge Bags con un valor de 551694 del tipo organic.


Y, como se observan para todas las variables coeficientes de varianza pequeños significa que los datos están cerca de su respectiva media y los valores de las variables no cambian mucho, entonces se podría decir que el mercado de aguacates es predecible en cuanto a precios y en cuanto a lo que produce.